controllers/autohealconfig_controller.go

```diff
@@ -47,16 +66,415 @@ type AutohealConfigReconciler struct {
+func (r *AutohealConfigReconciler) createConfigMaps(autoheal *v1alpha1.AutohealConfig, scenario []v1alpha1.AutohealConfigScenario, ctx context.Context) error {
+	log.Info(" Autoheal-operator, Stage: Creating  any new scenario config maps")
+
+	for _, scenarios := range scenario {
+		configmap := core.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      scenarios.Name,
+				Namespace: autoheal.Namespace,
+			},
+		}
+		query := scenarios.Query
+		label := scenarios.Label
+		script := scenarios.Script
+		graceperiod := scenarios.GracePeriod
+
+		_, err := controllerutil.CreateOrUpdate(ctx, r.Client, &configmap, func() error {
+			if configmap.Data == nil {
+				configmap.Data = make(map[string]string)
+			}
+			configmap.Labels = map[string]string{
+				"app": "autoheal",
+			}
+			configmap.Data["query"] = query
+			configmap.Data["label"] = label
+			configmap.Data["script"] = script
+			configmap.Data["gracePeriod"] = graceperiod
+			return nil
+
+		})
+		if err != nil {
+			log.Errorf("Unable to create config map")
+			return err
+
+		}
+	}
+	return nil
+}
+
+// Updates the controller to how long until next requeue of the reconciller depening on if its a error or not
+func (r *AutohealConfigReconciler) updateStatus(cr *v1alpha1.AutohealConfig) (ctrl.Result, error) {
+	err := r.Client.Status().Update(context.Background(), cr)
+	if err != nil {
+		return ctrl.Result{
+			Requeue:      true,
+			RequeueAfter: RequeueDelayError,
+		}, err
+	}
+
+	return ctrl.Result{
+		Requeue:      true,
+		RequeueAfter: RequeueDelaySuccess,
+	}, nil
+}
+
+// Delests the config secret with creds to sync to the repo
+func (r *AutohealConfigReconciler) deleteConfigSecret(config_cr *v1alpha1.AutohealConfig, ctx context.Context) error {
+	configRepoSecret := &core.Secret{}
+	secretSelector := &client.ObjectKey{
+		Namespace: config_cr.Namespace,
+		Name:      AutohealConfigSecret,
+	}
+	err := r.Client.Get(ctx, *secretSelector, configRepoSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Cannot find config map")
+
+		}
+		log.Errorf("Error cant get config repo secret")
+		return err
+
+	}
+	log.Info(" Autoheal-operator, Stage Clean up: Deleting config repo secret")
+	err = r.Client.Delete(ctx, configRepoSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Cannot find config secret")
+			return nil
+
+		}
+		log.Errorf("Error getting secret for deletion")
+		return err
+	}
+
+	return nil
+}
+
+// Delets all configs for cleanup
+func (r *AutohealConfigReconciler) deleteConfigmap(config_cr *v1alpha1.AutohealConfig, ctx context.Context) error {
+	configmap := &core.ConfigMapList{}
+	configSelector := &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"app": "autoheal",
+		}),
+		Namespace: config_cr.Namespace,
+	}
+
+	err := r.Client.List(ctx, configmap, configSelector)
+	if err != nil {
+		log.Errorf("Error cant get config repo secret")
```

Comment:
```suggestion
		log.Errorf("failed to get list of configmaps")
```
Looks like a copy/paste log line. The same thing is in the `removeUnwantedCM` function.